aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -231,17 +231,17 @@ struct NormZeroOps {
 #endif
 };
 
-template <typename acc_t>
+template <typename acc_t, typename data_t, typename out_t>
 struct NormOneOps {
-  inline C10_DEVICE acc_t reduce(acc_t acc, acc_t data, int64_t /*idx*/) const {
+  inline C10_DEVICE acc_t reduce(acc_t acc, data_t data, int64_t /*idx*/) const {
     return acc + std::abs(data);
   }
 
   inline C10_DEVICE acc_t combine(acc_t a, acc_t b) const {
     return a + b;
   }
 
-  inline C10_DEVICE acc_t project(acc_t a) const {
+  inline C10_DEVICE out_t project(acc_t a) const {
     return a;
```

Comment:
`a` is being passed as `acc_t`  type and will be returned as `out_t` type without any casting. will that work?

Reply:
`out_t` and `acc_t` is template parameter. If `acc_t` cannot be casted to `out_t`, it will raise error at compile time. But your comments makes a lot sense, we need to explicitly convert `acc_t` to `out_t`. I updated the code, could you help to review it again?